### ISSUE TITLE
Setting state inside rowEvents or using onSelect inside selectRow does not work

### ISSUE THREAD

**Issue by koookiii** [react-bootstrap-table/react-bootstrap-table2#246]
**Here's the error in the console:** 

`Uncaught TypeError: Cannot set property 'selected' of undefined
    at RowSelectionWrapper.componentWillReceiveProps (wrapper.js:64)
    at callComponentWillReceiveProps (react-dom.development.js:6389)
    at updateClassInstance (react-dom.development.js:6575)
    at updateClassComponent (react-dom.development.js:7848)
    at beginWork (react-dom.development.js:8225)
    at performUnitOfWork (react-dom.development.js:10224)
    at workLoop (react-dom.development.js:10288)
    at HTMLUnknownElement.callCallback (react-dom.development.js:542)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:581)
    at invokeGuardedCallback (react-dom.development.js:438)
    at renderRoot (react-dom.development.js:10366)
    at performWorkOnRoot (react-dom.development.js:11014)
    at performWork (react-dom.development.js:10967)
    at batchedUpdates (react-dom.development.js:11086)
    at batchedUpdates (react-dom.development.js:2330)
    at dispatchEvent (react-dom.development.js:3421)`

**Here's how my function is called:**
**( Logging the state inside rowEvents actually prints out the state so it's not a scope issue )**

```
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedRow: {}
    };
  }
  ...
  render(){
    const rowEvents = {
      onClick: (e, row, rowIndex) => {
        this.setState({ selectedRow: row });
      }
    };

  return(
 <BootstrapTable
          keyField="id"
          data={tableData}
          columns={columns}
          rowEvents={rowEvents}
          pagination={paginationFactory()}
          filter={filterFactory()}
          selectRow={selectRow}
          striped
          hover
          condensed
          />
       )
   }
}
```

**The same error pops up when you add** `onSelect` **to**` selectRow` **and handle the event with another function**

**Comment by gcfuji** [comment]
same error...

**Comment by loziobiz** [comment]
I'm getting the same error every time I try to use selectRow even in a ultra basic way like 

`const selectRow = {
  mode: 'checkbox',
  clickToSelect: true
};

<BootstrapTable keyField='id'
    data={_testdata}
    columns={columns}
    selectRow={selectRow} />
`

**Comment by Chun-MingChen** [comment]
Hi @koookiii 

Thanks for the information and it seems to be a bug :( We'll give an investigation further and fix it as soon as possible.

**Comment by karanbajaj2209** [comment]
Was it working in version 0.1.2 or 0.1.3?

**Comment by grogi** [comment]
@karanbajaj2209 The 0.1.2 is fine.

**Comment by Beerbossa** [comment]
Thanks @grogi

Yeah I had an issue where if I allowed row selection **and** changed the displayed data (array dynamically (located in the state of the parent component), then it would crash in a similar way. 0.1.3 seems to cover it 🤔 

Sick package though, really useful !

**Comment by AllenFang** [comment]
hey all, It's very sorry that we don't fix this critical issue in time, right now we fixed on `react-bootstrap-table-next@0.1.5`, please let me know if this issue still remain, thank

**Comment by Northshoot** [comment]
I am getting the same error when refetching or using state.selected

```
wrapper.js:64 Uncaught TypeError: Cannot set property 'selected' of undefined
    at RowSelectionWrapper.componentWillReceiveProps (wrapper.js:64)
    at callComponentWillReceiveProps (react-dom.development.js:6389)
    at updateClassInstance (react-dom.development.js:6575)
    at updateClassComponent (react-dom.development.js:7848)
    at beginWork (react-dom.development.js:8225)
```
